ironfish/src/network/peers/addressManager.ts

```diff
@@ -14,14 +14,28 @@ import { PeerManager } from './peerManager'
 export class AddressManager {
   hostsStore: HostsStore
   peerManager: PeerManager
+  peerIdentityMap: Map<string, PeerAddress>
```

Comment:
core change is a new data structure where we keep track of the peer candidates and attributes associated with them